src/ucp/core/ucp_request.c

```diff
@@ -395,50 +395,65 @@ static ucp_md_map_t ucp_request_get_invalidation_map(ucp_ep_h ep)
     return inv_map;
 }
 
-int ucp_request_memh_invalidate(ucp_request_t *req, ucs_status_t status)
+static UCS_F_ALWAYS_INLINE ucp_mem_h* ucp_request_get_memh(ucp_request_t *req)
```

Comment:
it seems like no need in pointer
```suggestion
static UCS_F_ALWAYS_INLINE ucp_mem_h ucp_request_get_memh(ucp_request_t *req)
```

Reply:
pointer is not needed as a return value of this function

Reply:
it's needed, please check again
We assign NULL value to the request field, which address is stored in `memh_p`

Reply:
ah.. found it but IMO this is a not good idea to set value using "getter", maybe reset the field directly by `req->send.state.dt_iter.type.contig.memh = NULL` at line 448?

Reply:
As you can see this getter returns different address depending on request type. It can be:
- `&req->send.state.dt_iter.type.contig.memh`
- `&req->send.state.dt.dt.contig.memh`
- plus in the new version `&req->send.rndv.mdesc->memh`

And the idea is to get this address once and then treat it uniformly.
I could set the field directly as you proposed, but then I would need to repeat the check

Reply:
maybe then add a helper function `ucp_request_extract_memh` like `ucp_wireup_extract_lane`?

Reply:
Thanks for the hint. That would work if we always unconditionally replace original memh field by NULL.
But it's slightly more involved here, because I use the same function in `ucp_request_memh_check_invalidate`, that must not change the request, so it must be pure get.
But I'll think a bit more on how to make it less confusing

Reply:
I refactored it in the latest PR with entire invalidation solution, used extract approach:
https://github.com/openucx/ucx/pull/10204/commits/03123f757fb58dccf12d48de4a7840c77c037867

Reply:
👍 does it mean that this PR is not relevant any more?
( just cross reference PRs https://github.com/openucx/ucx/pull/10204 )

Reply:
I was hoping to merge this small low-risk PR in the current release, but no luck.
I will confirm with tech leads on Monday whether it makes sense to split the overall solution into several pieces

Reply:
then maybe cherry-pick https://github.com/openucx/ucx/commit/03123f757fb58dccf12d48de4a7840c77c037867 to here?